### ISSUE TITLE
Copying .vim folder to iVim

### ISSUE THREAD

**Issue by ericweir** [terrychou/iVim#68]
I'm having a difficult time copying my .vim file to ivim. I've put a copy of my .vim folder in Dropbox. I rename the .vim folder in ivim to vim. Navigating to Dropbox in the Files app I select the folders in want to copy, navigate to the vim folder in ivim, and copy them there. then I rename the vim file in ivim to .vim.

I should say that's what I'm trying to do. One folder seems to hang up in the process and copies inconsistently. In my vim installation on my notebook I have used vundle to manage plugins. I have seven plugins installed. The plugin files are consolidated in a single "bundle" folder. 

This is the folder that is giving me trouble. It takes a long time. As I say, iIt seems to hang up. The plugin folders are copied incompletely and inconsistently.

Help appreciated. I've been away from vim for a while. Things that used to be intuitive no longer are.

**Comment by ericweir** [comment]
I've seen Working Copy mentioned here. Vundle requires git. I don't have git installed. Could that be the problem? Will Working Copy solve it?

**Comment by terrychou** [comment]
No, currently, iVim doesn't have git or related-function integrated, therefore you need to use a third party app, like Working Copy to do related work for you.

Moreover, iVim doesn't support running external commands either. As a result, any plugin which utilizes any external command would not work properly.

**Comment by ericweir** [comment]
Thanks, Terry. As mentioned in my original post, I'm coming back to vim from a few year's absence. My intuitions and skills are rusty. In addition, I'm a writer not a coder/programmer/hacker.

So, (1) should the method I describe in my original post for getting my .vim folder into iVim work? Do you have any idea why it seems to not be work for me? In particular, why everything except what's in the bundle folder copies quickly and without errors? Would you suggest and alternative method?

(2) I'm not familiar with the concept of "external commands." Restrictions on them aside, is installing Working Copy sufficient to make git available to plugins? If not what do I need to do to make them available?

(3) Do you know whether Vundle uses external commands? Vim-Plug is said to be a "minimalist" alternative to Vundle. Do you know whether it uses external commands? Do the restrictions on external commands make it unlikely that any plugin manager will work?

Thanks again.

**Comment by ericweir** [comment]
I see that vundle is a vim script. Does that mean it uses no external commands?

**Comment by terrychou** [comment]
1) the folder copying method you described above sounds OK to me. Since I don't even know what plugins you are trying to install, I have no idea why your way didn't work.

2) "external commands" are commands that are not part of vim itself. Examples include shell commands and other useful third party tools like git. Unlike other platforms on which a shell is available by default, iOS doesn't provide shell commands and users cannot install tools as easy as they do on a PC (installing Working Copy is not equivalent to installing git, it is just an app that provides git's function on iOS). Problems come when some plugins, in order to fulfill their functions, may call those "external commands" on their own, but they cannot find them (not available on iOS, remember?), therefore they stop working. There is not much you can do since it is a restriction from the platform.

3) Because Vundle supports to clone plugins from services like github, it at least needs the "external command" git. I think Vim-Plug does too. Plugin managers are just scripts that help you to maintain plugins. As long as they restrict their used functions within vim's internal ones, such as written in vim script, utilizing commands and functions provided by vim only, they would work in iVim.

Instead of using a plugin manager, I think a proper way for you to go first, is to try and install a plugin in the traditional way. (read the help document for plugin via ":help plugin" would be a good start.)

**Comment by ericweir** [comment]
Thanks, Terry. Much as I expected. But very helpful.

I had started thinking about manually installing plugins. I'll read up on it.

I did not use vundle that often. Just found it very convenient when I did. I have only a few plugins, and only three are critical to my day-to-day use: task paper, vim-sessions, and vimwiki. I hope when I eliminate all but these from my bundle folder I will simply be able to copy over the folders and files as installed in the .vim folder on my MacBook.

I don't know if manpageview is considered a plugin. I suspect it would help me to make more efficient and effective use of the vim manual/help. It comes as a vimball and should be easy to install.

Thanks again.

**Comment by ericweir** [comment]
I pared my vim plugins down to the four that are most critical for me. I put them in a "bundle" folder in the .vim folder. I added the following to my .vimrc:

“ http://www.vim.org/scripts/script.php?script_id=2027
“ https://github.com/xolox/vim-misc
“ https://github.com/xolox/vim-session
“ https://github.com/vimwiki/vimwiki

set runtimepath^=~/.vim/bundle/taskpaper.vim/taskpaper.vim
set runtimepath^=~/.vim/bundle/vim-misc/misc.vim
set runtimepath^=~/.vim/bundle/vim-session/session.vim
set runtimepath^=~/.vim/bundle/vimwiki/vimwiki.vim

I am uncertain whether it is sufficient to declare the folder containing all the plugin folders, as above, or whether the subfolder containing the vimscript for the plugin should also be declared, as in the following example:
  
set runtimepath^=~/.vim/bundle/vimwiki/plugin/vimwiki.vim

Actually, I'm uncertain about more than that, but that's the question that concerns me most at the moment.

**Comment by ericweir** [comment]
I noticed a request here to upgrade ivim to vim 8. An advantage would be that the built-in plugin manager could be used.

**Comment by ericweir** [comment]
Assuming my installation of plugins is correct--they show up in my runtime path--I'm finding I apparently don't know how to generate help files for them. The following is an example of the command I've used:

:helptags ~/.vim/bundle/taskpaper.vim/doc

But when I do ":help task paper" I keep getting ":149: Sorry, no help for taskpaper."

What am I missing/doing wrong?

Thanks.

**Comment by terrychou** [comment]
It may be a problem of the searching path of the help tags. 

Here is one paragraph from help on help:

> When no argument is given to |:help| the file given with the 'helpfile' option
will be opened.  Otherwise the specified tag is searched for in all "doc/tags"
files in the directories specified in the 'runtimepath' option.

**Comment by ericweir** [comment]
Thanks, Terry. Apologies for the late response. I've been distracted.

Specification of the runtimepath was indeed the problem, e.g., /.vim/bundle/vimwiki/vimwiki.vim should have been /.vim/bundle/vimwiki

Now onto defamiliarizing myself with my plugins/
